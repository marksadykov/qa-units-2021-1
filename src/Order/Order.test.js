jest.mock('../utils/getDate');

import React from 'react'
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Order from "./Order";
import {getDate} from "../utils/getDate";

Enzyme.configure({ adapter: new Adapter() })

describe('Order.js', () => {
  getDate.mockReturnValue('9 декабря, вс, 2018 год');

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Order with props', () => {

    const order = {
      date: 1544356800000,
      shop: 'text',
      items: ['1', '2']
    };

    const output = shallow(
        <Order order={order} />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
    expect(getDate).toHaveBeenCalledTimes(1);
  });

  it('order.items is undefined', () => {
    const order = {
      date: 1544356800000,
      shop: 'text',
      items: undefined,
    };

    const output = shallow(
        <Order order={order} />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
    expect(getDate).toHaveBeenCalledTimes(1);
  });

  it('order.items have 0 length', () => {
    const order = {
      date: 1544356800000,
      shop: 'text',
      items: [],
    };

    const output = shallow(
        <Order order={order} />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
    expect(getDate).toHaveBeenCalledTimes(1);
  });

  it('order is undefined', () => {
    const order = undefined;

    const output = shallow(
        <Order order={order} />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('order.shop is undefined', () => {
    const order = {
      shop: undefined,
    };

    const output = shallow(
        <Order order={order} />
    );
    expect(shallowToJson(output)).toMatchSnapshot()
  });

  it('order.date is undefined', () => {
    const order = {
      date: undefined,
    };

    const output = shallow(
        <Order order={order} />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('order.shopa and order.date are undefined', () => {
    const order = {
      date: undefined,
      shop: undefined,
    };

    const output = shallow(
        <Order order={order} />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('render without props', () => {
    const output = shallow(
        <Order />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});

