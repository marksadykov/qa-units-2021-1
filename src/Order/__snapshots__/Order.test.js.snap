// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order.js Order with props 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      text
    </span>
    <span
      id="date"
    >
      9 декабря, вс, 2018 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      1
    </div>
    <div
      className="Order-item"
      key="1"
    >
      2
    </div>
  </div>
</div>
`;

exports[`Order.js order is undefined 1`] = `""`;

exports[`Order.js order.date is undefined 1`] = `""`;

exports[`Order.js order.items have 0 length 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      text
    </span>
    <span
      id="date"
    >
      9 декабря, вс, 2018 год
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;

exports[`Order.js order.items is undefined 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      text
    </span>
    <span
      id="date"
    >
      9 декабря, вс, 2018 год
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;

exports[`Order.js order.shop is undefined 1`] = `""`;

exports[`Order.js order.shopa and order.date are undefined 1`] = `""`;

exports[`Order.js render without props 1`] = `""`;
